app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java

```diff
@@ -68,21 +71,64 @@ protected void onBindViewHolder(@NonNull BookListViewHolder holder, int position
         bookListViewHolder.author.setText(book.getAuthor());
         bookListViewHolder.isbn.setText(book.getIsbn());
 
-        book.setStatusCircleColor(book.getStatus(), bookListViewHolder.statusCircle);
+        switch(this.caller) {
+            case R.id.my_books:
+                book.setStatusCircleColor(book.getStatus(), bookListViewHolder.statusCircle);
+
+                holder.itemView.setOnClickListener(new View.OnClickListener()
+                {
+                    // Handles a click on an item in the recycler view
+                    @Override
+                    public void onClick(View v)
+                    {
+                        // Opens the book in detail view
+                        MyBooksDetailViewFragment bookDetailFragment = new MyBooksDetailViewFragment();

```

Comment:
All of these functions are essentially the same. Is it possible to extract to a function and pass in a parameter of type Fragment?